rustls/src/msgs/deframer.rs

```diff
@@ -504,7 +504,8 @@ impl DeframerBuffer<'_, InternalPayload> for DeframerVecBuffer {
 
 #[cfg(feature = "std")]
 impl<'a> DeframerBuffer<'a, ExternalPayload<'a>> for DeframerVecBuffer {
-    fn copy(&mut self, payload: &ExternalPayload<'a>, _at: usize) {
+    fn copy(&mut self, payload: &ExternalPayload<'a>, at: usize) {
+        debug_assert_eq!(at, self.used);
```

Comment:
You [also indicated](https://github.com/rustls/rustls/pull/1798#discussion_r1583331582) you thought some of the context from the fix commit message deserved to be a comment. That sounds reasonable to me, could you roll that into this diff?

Reply:
Yes, I was thinking about that too. However it is not trivial how to formulate this context from a commit message explaining why this was changes into a more permanent why is it "ok to be like this" comment for the code.
While thinking about it - I figures that maybe this is actually not an ok thing to have - and a broader remodelling of how this trait and in general part of logic is in order.

If you can agree this is the case - that is what I'll capture and add as a comment.